### PR TITLE
Update New-AzADUser.md

### PR DESCRIPTION
**Pull request by theheatDK**
-MailNickname is required but cannot contain @.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources/help/New-AzADUser.md [hunk 1]
```diff
@@ -14,8 +14,7 @@ Creates a new active directory user.
 ## SYNTAX
 
 ```
-New-AzADUser -DisplayName <String> -UserPrincipalName <String> -Password <SecureString> [-ImmutableId <String>]
- [-MailNickname <String>] [-ForceChangePasswordNextLogin] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
+New-AzADUser -DisplayName <String> -UserPrincipalName <String> -Password <SecureString> -MailNickname <String> [-ImmutableId <String>] [-ForceChangePasswordNextLogin] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/help/New-AzADUser.md]: This parameter is not required in the cmdlet code - are you sayign that it is actually required in some circumstances, or is this an oversight?

Comment (theheatDK) [reply src/ResourceManager/Resources/Commands.Resources/help/New-AzADUser.md]: I get an error if I do not provide it:
```
PS C:\svn\obs\azure\scripts> $SecureStringPassword = ConvertTo-SecureString -String "TxQl77FS*#m9" -AsPlainText -Force
New-AzADUser -DisplayName "dnhf01-test01" -UserPrincipalName "dnhf01-test03@xyz.onmicrosoft.com" -Password $SecureStringPassword
New-AzADUser : 'MailNickname' cannot be null.
At line:2 char:1
+ New-AzADUser -DisplayName "dnhf01-test01" -UserPrincipalName "dnhf01- ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzADUser], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ActiveDirectory.NewAzureADUserCommand
```

Comment (theheatDK) [reply src/ResourceManager/Resources/Commands.Resources/help/New-AzADUser.md]: I use version 1.0.1:
```
PS C:\svn\obs\azure\scripts> Get-Module Az

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.0.1      Az
```

**Comment by markcowl** [comment]
@theheatDK So in that case, there is a coding issue, which will require more than a help fix - this cmdlet should be marked as required if it is actually required.

File: src/ResourceManager/Resources/Commands.Resources/help/New-AzADUser.md (hunk 1 above)
Comment (markcowl) [reply src/ResourceManager/Resources/Commands.Resources/help/New-AzADUser.md]: So in that case, there is a coding issue, which will require more than a help fix - this cmdlet should be marked as required if it is actually required.

**Comment by markcowl** [comment]
@cormacpayne, can you look at this and dtermine if the parameter sets need to be updated, or the cmdlet implentation needs to be resilient over no MailNickname provided?

**Comment by maddieclayton** [comment]
Closing this PR in favor of issue opened: https://github.com/Azure/azure-powershell/issues/8220.  We will look into the cmdlet code and figure out whether this parameter should be required, or if there is some other issue with the cmdlet code.  Please add any further information or questions to the issue.  Thanks @theheatDK for pointing out this issue.

**Pull request closed without merging**